sage/pallet/src/mock.rs

```diff
@@ -182,57 +161,26 @@ impl SeasonManager for MockSeasonManager {
 }
 
 pub struct TestSageEngine;
-// The macro can't handle the brackets.
-type TestBlockNumber = BlockNumberFor<Test>;
-
-/// Runtime specific sage implementation so that we don't have to
-/// pass our type definitions all the time.
-macro_rules! impl_runtime_sage_api {
-	(
-		$impl_target:ident,
-		$runtime:ident,
-		$sage_instance:ident,
-		$season_manager:ident,
-		$asset_id:ident,
-		$asset:ident,
-		$transition_config:ident,
-	) => {
-		impl_sage_api!(
-			$impl_target,
-			$runtime,
-			$sage_instance,
-			$season_manager,
-			Randomness,
-			MockAccountId,
-			$asset_id,
-			$asset,
-			FungiblesAssetId,
-			MockBalance,
-			TestBlockNumber,
-			MockSeasonId,
-			$transition_config,
-			H256,
-		);
-	};
-}
 
 // Every new game we add can simply call that macro for another sage instance to
 // implement the sage api given that the other types are identical.
-impl_runtime_sage_api!(
+impl_test_runtime_sage_api!(
 	TestSageEngine,
 	Test,
 	DefaultSageInstance,
 	MockSeasonManager,
 	AssetId,
 	MockAsset,
 	CasinoJamTransitionConfig,
+	Randomness,
```

Comment:
Here, we could inject the `TestRandomness` from sage-testing, however, we have a few tests here that ensure that the randomness is indeed random, so I chose to keep the actual randomness. The test randomness can be used when testing the game transitions.